distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2112,6 +2144,184 @@ public void testProductDownloadWithTwoUsers() throws Exception {
                 "COMPLETE");
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // where 3 is the number of notifications received by the cometD client from the
+        // notifications channel and 9 is the number of activities received by the cometD client
+        // from the activities channel
+        checkExpectations(cometDClient1, 3, 9);
+        checkExpectations(cometDClient2, 3, 9);
+
+        List<String> activities = cometDClient1.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        boolean activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        activities = cometDClient2.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        List<String> notifications = cometDClient1.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        boolean notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+        notifications = cometDClient2.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+    }
+
+    @Test
+    public void testTwoUsersSameProductRetryFailure() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForFailure(cometDClient1, filename);
+        setupCswServerForFailure(cometDClient2, filename);
+
+        // where 4 is the number of notifications received by the cometD client from the
```

Comment:
@coyotesqrl the notifications and activities are tightly coupled to cometD, we can't predict if an update to cometD will change the notifications/activities in the future